modules/apps/27-interchain-accounts/interchain_accounts_test.go

```diff
@@ -0,0 +1,119 @@
+package interchain_accounts_test
+
+import (
+	"github.com/cosmos/ibc-go/v2/modules/apps/27-interchain-accounts/types"
+	channeltypes "github.com/cosmos/ibc-go/v2/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/v2/modules/core/24-host"
+)
+
+// Test initiating a ChanOpenInit using the host chain instead of the controller chain
+// ChainA is the controller chain. ChainB is the host chain
+func (suite *InterchainAccountsTestSuite) TestChanOpenInitWrongFlow() {
+	suite.SetupTest() // reset
+	path := NewICAPath(suite.chainA, suite.chainB)
+	suite.coordinator.SetupConnections(path)
+
+	// use chainB (host) for ChanOpenInit
+	msg := channeltypes.NewMsgChannelOpenInit(path.EndpointB.ChannelConfig.PortID, types.VersionPrefix, channeltypes.ORDERED, []string{path.EndpointB.ConnectionID}, path.EndpointA.ChannelConfig.PortID, types.ModuleName)
+	handler := suite.chainB.GetSimApp().MsgServiceRouter().Handler(msg)
+	_, err := handler(suite.chainB.GetContext(), msg)
+
+	suite.Require().Error(err)
+}
+
+// Test initiating a ChanOpenTry using the controller chain instead of the host chain
+// ChainA is the controller chain. ChainB creates a controller port as well,
+// attempting to trick chainA.
+func (suite *InterchainAccountsTestSuite) TestChanOpenTryWrongFlow() {
+	suite.SetupTest() // reset
+	path := NewICAPath(suite.chainA, suite.chainB)
+	suite.coordinator.SetupConnections(path)
+
+	err := InitInterchainAccount(path.EndpointA, TestOwnerAddress)
+	suite.Require().NoError(err)
+
+	// chainB also creates a controller port
+	err = InitInterchainAccount(path.EndpointB, TestOwnerAddress)
+	suite.Require().NoError(err)
+
+	path.EndpointA.UpdateClient()
+	channelKey := host.ChannelKey(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
+	proofInit, proofHeight := path.EndpointB.Chain.QueryProof(channelKey)
+
+	// use chainA (controller) for ChanOpenTry
+	msg := channeltypes.NewMsgChannelOpenTry(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, TestVersion, channeltypes.ORDERED, []string{path.EndpointA.ConnectionID}, path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, types.VersionPrefix, proofInit, proofHeight, types.ModuleName)
+	handler := suite.chainA.GetSimApp().MsgServiceRouter().Handler(msg)
+	_, err = handler(suite.chainA.GetContext(), msg)
+
+	suite.Require().Error(err)
+}
+
+// Test initiating a ChanOpenAck using the host chain instead of the controller chain
+// ChainA is the controller chain. ChainB is the host chain
+func (suite *InterchainAccountsTestSuite) TestChanOpenAckWrongFlow() {
+	suite.SetupTest() // reset
+	path := NewICAPath(suite.chainA, suite.chainB)
+	suite.coordinator.SetupConnections(path)
+
+	err := InitInterchainAccount(path.EndpointA, TestOwnerAddress)
+	suite.Require().NoError(err)
+
+	err = path.EndpointB.ChanOpenTry()
+	suite.Require().NoError(err)
+
+	// chainA maliciously sets channel to TRYOPEN
```

Comment:
Is the idea that by doing this a relayer will try and call `OnChanOpenAck` on the host chain as the next step? What would be the benefit of this to the malicious chain A? 

Just trying to better understand what we are testing here (other than the sanity check) :)

Reply:
> Is the idea that by doing this a relayer will try and call `OnChanOpenAck` on the host chain as the next step?

The idea is that the relayer could call OnChanOpenAck on the controller chain using the host port ID. 

>  What would be the benefit of this to the malicious chain A?

The host portID would be marked as an active channel. I'm not sure if there is any benefit. This is mostly just a sanity check, the existing logic makes it difficult to confuse a host or controller chain